### PR TITLE
Test & refactor the Store write logic

### PR DESCRIPTION
**Pull request by Cgg**
- extract the output directory from the module and pass it as an argument
- test the `newBundle` function

### CODE REVIEW

File: src/store/index.ts [hunk 1]
```diff
@@ -32,36 +28,55 @@ const writeToDisk = (b: Buffer | string, path: string): Promise<void> =>
...
-export const writeOne = async (a: File.newFile): Promise<File.File> => {
-  const dir = config?.storage?.filesystem?.directory;
-  if (!dir) {
-    throw new Error(`No directory set in config`);
-  }
+const writeOne = async (
+  newFile: File.NewFile,
+  configuration: WriteConfiguration
+): Promise<File.File> => {
+  const { directory } = configuration;
```
Comment (basilesimon) [inline src/store/index.ts:45]: Even though `@param configuration: WriteConfiguration` is typed, wouldn't we set `directory` conditionally, e.g. with [optional chaining](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining)?

Right now, yeah `directory` is required but I can see this file rewritten with partial applications for different locations, e.g.

```ts
const writeOne = (config: Configuration, f: File.NewFile) => { ... };
const writeLocalFS = partial(writeOne, { directory: 'out/' });
const writeS3 = partial(writeOne, { bucket: 's3://archives' });
```

Don't really what's the right answer there, if I'm honest. This file has deep roots in its own configuration, it feels kind of odd parameterising it like this, if you see what I mean?

Comment (Cgg) [reply src/store/index.ts:45]: We can alter `Configuration` when this becomes necessary; with TS and test the refactoring should be straightforward. Until then keeping things simple looks to me like the better option.

Comment (Cgg) [reply src/store/index.ts:45]: I'd also probably keep things simple and currying out of the equation, eg
```ts
type Configuration = {
  s3?: ...
  directory?: ...
};

function writeOne(nf: NewFile, { s3, directory }: Configuration) {
  if (s3 === undefined && directory === undefined) {
    throw new Error('At least one write option must be specified');
  }
  if (s3) {
    writeToS3(nf, s3);
  }
  if (directory) {
    writeToDisk(nf, directory);
  }
}
```

I should make `writeToDisk` take only the relevant parameter (ie the directory path), let me do that.

Comment (Cgg) [reply src/store/index.ts:45]: Looking at this again, much of what is `writeOne` is about writing to disk, and thus should be moved to `writeToDisk`. I'll do that presently.

**Review by basilesimon — APPROVED** [review]

**Pull request merged**